lib/tasks/asset_sync.rake

```diff
@@ -20,11 +20,24 @@ if Rake::Task.task_defined?("assets:precompile:nondigest")
     Rake::Task["assets:sync"].invoke if defined?(AssetSync) && AssetSync.config.run_on_precompile
   end
 else
+  # Triggers on webpacker compile task instead of assets:precompile to ensure
```

Comment:
@jbusser I took this exactly from AssetSync/asset_sync#341, but I also considered this simpler approach:

```ruby
if Rake::Task.task_defined?("webpacker:compile")
  Rake::Task["webpacker:compile"].enhance do
    if defined?(AssetSync) && AssetSync.config.run_on_precompile
      Rake::Task["assets:sync"].invoke
    end
  end
else
  Rake::Task["assets:precompile"].enhance do
    if defined?(AssetSync) && AssetSync.config.run_on_precompile
      Rake::Task["assets:sync"].invoke
    end
  end
end
```

It seems strange to me to check `AssetSync.config.include_webpacker_assets`, since we're already doing that inside of `assets:sync`. I also feel like it could get us into a situation where assets just don't get synced at all if `Rake::Task.task_defined?("webpacker:compile")` returns false, but `AssetSync.config.include_webpacker_assets` returns true (see line 39).

Anyway, I think it may be better to just keep this part simple to focus on which task we're enhancing and make sure assets get synced either way. Let me know what you think!